### ISSUE TITLE
Doesn't properly handle variable clock Skylake CPUs

### ISSUE THREAD

**Issue by justinkterry** [lpereira/hardinfo#60]
Current CPUs have min/max clock speeds that will be adjusted based on load. Benchmarking tools should always switch the processor to the highest clock speed (turbo) to properly determine performance. Using Ubuntu Gnome 17.04 on a Skylake series Intel i7-6700k CPU, this behavior does not occur. The CPU performance test all ran my CPU at the lowest possible clock speed. Please contact me if you have any questions or would like me to test something.

**Comment by mckaygerhard** [comment]
the switch process in that CPU are based on load, and the real problem here its that the application does not represent a "necesary to stepping" to the CPU.. that due  the adoption of technologies that naturally prove speedier in certain applications. Yes, you get the expected notched-up chipset I/O throughput, as well as nods to more familiar technologies like DDR4 memory. This means that the CPU does not has spected behaviour.. and obviously its a windoze CPU due the intel drivers and utilities combinatios its the real stepping manager... as example the max base clock detecte was 127Mhz and the specifications said more, if a windoxe app reported that u spected, its due are hardcoded, trying olders version of same app reports differends numbers..

from my view, u have a fake CPU, that only works with specific things.. desktop Skylake doesn't present as compelling a story as we've seen from Intel's desktop chips in the past—another sign of how the chip world is rapidly moving away from the segment that has traditionally been its power king.

sorry for u, but the industry the only thing that wants its your money... the only CPU that hav a real ondeman its the quad-core 2.7GHz Core i7-6820HK CPU with an unlocked multiplier..

so in conclusion, application (hardinfo) does not have any control over the cpu, so application can handle the stepping change pver the CPU, and due Linux based OS are so good performance, that process does not represent a load charge.. this with that that CPU its made with specific apps (wuindoze like).. u dont have a right chooce... that u spected

**Comment by inferno596** [comment]
But, at least user password authentication, it should be easily doable to switch the CPU into turbo for the tests. That's built into Linux already. Without that the bench marks aren't uniform or standardized between CPUs for bench marks as short as the ones included in hardinfo. 

Your claim that CPUs like this are windows exclusive is also unreasonable and completely wrong. High performance Intel CPUs dominate the server market, which is also dominated by Linux, hence Intel's massive open driver development team. Linux has excellent support for all modern Intel CPUs, and that's what the bulk of all the most recent kernel updates have concerned.

**Comment by mckaygerhard** [comment]
> But, at least user password authentication, it should be easily doable to switch the CPU into turbo for the tests. That's built into Linux already.

if u lauch hardinfo as normal user, of course will not ask anything, due the test are made after application behavior was determined.. so that u said its nonsense, try lauch the app as root, of course that is a non-sense way due for root right usage u must try a console based command tools by separate.. 

> Without that the bench marks aren't uniform or standardized between CPUs for bench marks as short as the ones included in hardinfo.

hardinfo as i know its made for light desktops and as informative tool in user level, for more u must use the proper desktop related tool or a advanced tool in low level or user-space level, the behaviour of the intel and enterprise made hardware are not in sync with community projects so each developer could (and not) be a "intel suck di-ck".. its a long history of non or limited poor contribution with linux and GNU community

> Your claim that CPUs like this are windows exclusive is also unreasonable and completely wrong. High performance Intel CPUs dominate the server market, which is also dominated by Linux, hence Intel's massive open driver development team. Linux has excellent support for all modern Intel CPUs, and that's what the bulk of all the most recent kernel updates have concerned.

Intel have support for linux, and have open specifications, but as i already said, that intel cpu its not server focus, so that u said are not right at all, intel collaborate only with linux on more close to server market, on desktop its a windolike parner.. specially on gamming.. and this cpu have very GPU features for windogamers,..

and in final, and in this u may have right, the test are very poor load respect the high power of the cpu.. so maybe some low level code must be made.. but many aspecs are in the middle way.. 
ah, final note: that cpu seem was with some bugs that microcode releases fixed..

**Comment by lpereira** [comment]
If you're using an on-demand governor, the CPU will quickly ramp up to a high clock frequency when doing CPU-bound work.  The benchmarks in HardInfo might not be enough to make the CPU go all the way up to 11, though, so you might want to use the `cpuset(7)` utility (usually `sudo cpuset frequency-set -g performance` will do the trick).

Another thing that you'll notice is that, even when using the performance governor, HardInfo won't update the CPU frequency information.  This is misleading but it's a known limitation.

**Comment by inferno596** [comment]
I understand how to get around the limitations, I was just wondering if anyone was going to fix this given that it's a known bug in proper performance representation and this is a major Ubuntu module.

**Comment by mckaygerhard** [comment]
its a complicted to said if are a limitation or now, and maybe not due its not really a bug.. due this application u are runny at the user level.. and as i know, in real linux world a administrator know wnat must be done.. as author just said, runs the commands and u have it..

if any code will be made, must determine the cpu type, then run a special test, or works, must made the ticks for frecuecy clock and then run the tests.. too complicated only for a intel bad desing made only for windolike gammers.. 

if u have a code with a pull request, good, if not use a winbuntu tool

**Comment by lpereira** [comment]
@inferno596 It's possible this will get fixed, yes.  But keep in mind that work in HardInfo has stopped for a while; I'm mostly maintaining it.

**Comment by bp0** [comment]
This could be considered fixed by 7f2006d10ec45dc72f9373327b632d167d37cb07.

**Comment by lpereira** [comment]
Closing issue as it's been fixed.
